src/main/java/us/ihmc/commons/thread/RepeatingTaskThread.java

```diff
@@ -416,31 +409,34 @@ private synchronized void addScheduled(long repetitions)
        *
        * @return {@code true} if interrupted.
        */
-      private synchronized boolean waitForChange()
+      private boolean waitForChange()
       {
-         try
+         synchronized (this)
```

Comment:
Why not have the method be synchronized? A synchronized method is synchronized on this object anyway

Reply:
reverted